2.4/root/usr/share/container-scripts/mongodb/common.sh

```diff
@@ -134,39 +134,41 @@ function replset_addr() {
 }
 
 # mongo_remove removes the current MongoDB from the cluster
+# optionally takes identifier of replset to connect and address of server to remove
 function mongo_remove() {
   local host
   # if we cannot determine the IP address of the primary, exit without an error
   # to allow callers to proceed with their logic
-  host="$(replset_addr || true)"
+  host="${1:-$(replset_addr || true)}"
+  local mongo_addr
+  mongo_addr="${2:-$(mongo_addr)}"
```

Comment:
> In this case it would have a different behavior, it would set host to the empty string (and not call replset_addr)...

@rhcarvalho Really? `:-` is "If parameter is unset or null"

> And you were arguing there that we should use the replica set addressing :-)
And I agree with you!

Yes i think we should. I am trying to convice @php-coder. But now it is better for me to have more generic function than do not use this function.

> That's why I would defer this to be worked on #184, not as a separate PR since it has no clear benefits without that other one (which is work in progress...).

I agree.
Too late? :-)